Description: Enable SSE2 on i386 and amd64 CPUs only
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
Taken from: https://sources.debian.org/patches/openjfx/11.0.2+1-1/08-disable-sse2.patch
--- old/modules/javafx.media/src/main/native/gstreamer/projects/linux/avplugin/Makefile
+++ new/modules/javafx.media/src/main/native/gstreamer/projects/linux/avplugin/Makefile
@@ -24,13 +24,16 @@
          -fstack-protector       \
          -Werror=implicit-function-declaration \
          -Werror=trampolines     \
-         -msse2                  \
          -fbuiltin               \
          -DHAVE_STDINT_H         \
          -DLINUX                 \
          -DGST_DISABLE_LOADSAVE  \
          -DGSTREAMER_LITE \
          -ffunction-sections -fdata-sections
+
+ifneq (,$(filter $(CARCH), x86 x86_64))
+    CFLAGS += -msse2
+endif
 
 ifeq ($(BUILD_TYPE), Release)
     CFLAGS += -Os
--- old/modules/javafx.media/src/main/native/gstreamer/projects/linux/fxplugins/Makefile
+++ new/modules/javafx.media/src/main/native/gstreamer/projects/linux/fxplugins/Makefile
@@ -21,7 +21,6 @@
          -fstack-protector       \
          -Werror=implicit-function-declaration \
          -Werror=trampolines     \
-         -msse2                  \
          -fbuiltin               \
          -DHAVE_STDINT_H         \
          -DLINUX                 \
@@ -32,6 +31,10 @@
          -DGST_DISABLE_GST_DEBUG \
          -DGSTREAMER_LITE \
          -ffunction-sections -fdata-sections
+
+ifneq (,$(filter $(CARCH), x86 x86_64))
+    CFLAGS += -msse2
+endif
 
 ifeq ($(BUILD_TYPE), Release)
     CFLAGS += -Os
--- old/modules/javafx.media/src/main/native/jfxmedia/projects/linux/Makefile
+++ new/modules/javafx.media/src/main/native/jfxmedia/projects/linux/Makefile
@@ -43,8 +43,11 @@
                   -fstack-protector \
                   -Werror=implicit-function-declaration \
                   -Werror=trampolines \
-		  -msse2 \
 	          -DGSTREAMER_LITE
+
+ifneq (,$(filter $(CARCH), x86 x86_64))
+	CFLAGS += -msse2
+endif
 
 	PACKAGES_INCLUDES := $(shell pkg-config --cflags glib-2.0)
 	PACKAGES_LIBS := $(shell pkg-config --libs glib-2.0 gobject-2.0 gmodule-2.0 gthread-2.0)
